### PR TITLE
Set default language to first element of LANGS

### PR DESCRIPTION
**Pull request by 6543**
The first element of `LANGS` in `[i18n]` section will be teh default language.
Witch is used if no language is set to translate strings.

### CODE REVIEW

**Comment by techknowlogick** [comment]
CI fail is related.

**Comment by zeripath** [comment]
I don't get it. LANGS when would you want the default not to be the first element of that list?

File: modules/translation/translation.go [hunk 1]
```diff
@@ -38,6 +36,7 @@ func AllLangs() []LangType {
 
 // InitLocales loads the locales
 func InitLocales() {
+	i18n.Flush()
```
Comment (lunny) [inline modules/translation/translation.go]: The change seems unrelated with the topic?

Comment (6543) [reply modules/translation/translation.go]: Dont think it's worth its own pull

File: modules/setting/setting.go [hunk 2]
```diff
@@ -944,6 +945,7 @@ func NewContext() {
 			"español", "português do Brasil", "Português de Portugal", "polski", "български",
 			"italiano", "suomi", "Türkçe", "čeština", "српски", "svenska", "한국어"}
 	}
+	DefaultLang = Cfg.Section("i18n").Key("Default").MustString("en-US")
```
Comment (lafriks) [inline modules/setting/setting.go]: Default language should be first language from list otherwise it will be breaking change

**Review by lafriks — CHANGES_REQUESTED** [review]

File: modules/translation/translation.go (hunk 1 above)
Comment (6543) [reply modules/translation/translation.go]: will make a own pull ...

Comment (6543) [reply modules/translation/translation.go]: ->  #15073

**Comment by zeripath** [comment]
I still don't understand this PR. The Langs setting is a list - when would you not want the default to be the first element of that list and why?

**Comment by 6543** [comment]
my first approach was a new Key ... **lafriks** wrote: https://github.com/go-gitea/gitea/pull/14846#discussion_r589766331

and why:
At the moment if a user has no lang set, it will be english - if this is a Installation focused on a specific user base not that familiar with english, the default can not be changed

**Review by zeripath — APPROVED** [review]

**Comment by lafriks** [comment]
I don't think that's correct currently you are changing not default user language but fallback language that is not guaranteed to be 100% completed if you change default user language to something other

**Comment by 6543** [comment]
you are right :/ ... I should call it Fallback and change it in this direction

**Comment by lafriks** [comment]
But is this really needed? What will happen if fallback language is missing translations?

**Comment by 6543** [comment]
> But is this really needed? What will happen if fallback language is missing translations?

-> as you suggested it's english again

**Comment by 6543** [comment]
As per discord: since this only cover edge case, just go with eng
